cpp/src/io/json/nested_json.hpp

```diff
@@ -216,13 +231,13 @@ struct json_column {
     // Struct   | List      => FAIL
```

Comment:
Where does this "FAIL" happen?

Reply:
It fails in the child column selection but let me add a more explicit message for these two cases

Reply:
Thanks for pointing out, I've added explicit check and error message and test for this case!